hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterFileSystem.java

```diff
@@ -331,16 +331,15 @@ private void checkSubDir(final Path p, final String dirPermsConfName) throws IOE
           throw new IOException("HBase directory '" + p + "' creation failure.");
         }
       }
-    } else {
-      if (isSecurityEnabled && !dirPerms.equals(fs.getFileStatus(p).getPermission())) {
-        // check whether the permission match
-        LOG.warn("Found HBase directory permissions NOT matching expected permissions for "
-          + p.toString() + " permissions=" + fs.getFileStatus(p).getPermission() + ", expecting "
-          + dirPerms + ". Automatically setting the permissions. "
-          + "You can change the permissions by setting \"" + dirPermsConfName
-          + "\" in hbase-site.xml " + "and restarting the master");
-        fs.setPermission(p, dirPerms);
-      }
+    }
+    if (isSecurityEnabled && !dirPerms.equals(fs.getFileStatus(p).getPermission())) {
```

Comment:
This part looks good because here we check permission regardless of how/when dir was created.